Fix build with opaque EVP_MD_CTX and HMAC_CTX in LibreSSL 3.5.

Index: src/lua/lua_cryptobox.c
--- src/lua/lua_cryptobox.c.orig
+++ src/lua/lua_cryptobox.c
@@ -993,7 +993,7 @@ static void
 lua_cryptobox_hash_dtor (struct rspamd_lua_cryptobox_hash *h)
 {
 	if (h->type == LUA_CRYPTOBOX_HASH_SSL) {
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 		EVP_MD_CTX_cleanup (h->content.c);
 #else
 		EVP_MD_CTX_reset (h->content.c);
